### DISCUSSION TITLE
如何自定义国际化

### DISCUSSION THREAD

**Discussion by MissNanLan** [alibaba/designable#89] [Q&A] [answered]
背景： 因为公司使用了[react-i18next](https://react.i18next.com/)，为了统一，我们不打算走`designable`本身的提供`designerLocales`的参数

具体步骤：

1 、改变切换入口

![image](https://user-images.githubusercontent.com/22790863/136538737-38de4ea2-f757-4d85-8bc9-46b80f1e1378.png)

2、 使用hook

使用 hooks [ useTranslation](https://react.i18next.com/latest/usetranslation-hook) 

![image](https://user-images.githubusercontent.com/22790863/136538603-40be6186-3c4a-41cd-99cb-184baf25713c.png)

以上有作用

但是在propsSchmea里面配置

![image](https://user-images.githubusercontent.com/22790863/136538910-a26e84ee-bda4-433b-8ba8-e13faebaae85.png)

中英文切换不起任何效果

审查代码，只要把t函数通过SettingForm的scope传入，再去改下react-settings-form里面的SettingsForm的源码

`SettingForm` 传入`scope`参数

![image](https://user-images.githubusercontent.com/22790863/136539149-9e01f6ff-df3a-4078-87ed-b266c34128f8.png)

`react-settings-form`里面的SettingsForm加入 `props?.scope`

![image](https://user-images.githubusercontent.com/22790863/136539308-7189c035-f183-4dd9-82a7-81ff390c3fb4.png)

这样就可以有作用

由于codesandbox里面限制500个`node_modules`，只能提供一部分代码[https://codesandbox.io/s/elastic-scott-32dwr](https://codesandbox.io/s/elastic-scott-32dwr)

如果有其它更好的思路，希望能告知，感谢

**Comment by janryWang** [comment, marked as answer]
推荐把你们自己的locale language用observable.ref包一下，这样就能响应变化了，而不是整体受控模式

**Reply by MissNanLan** [reply]
好的我试试
